helper-cli/src/main/kotlin/commands/SubtractScanResultsCommand.kt

```diff
@@ -0,0 +1,119 @@
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.mapper
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.utils.expandTilde
+
+internal class SubtractScanResultsCommand : CliktCommand(
+    help = "Subtracts the given right-hand side scan results from the given left-hand side scan results. The output " +
+            "is written to the given output ORT file."
+) {
+    private val lhsOrtResultFile by option(
+        "--lhs-ort-result-file",
+        help = "The ORT result containing the left-hand-side scan result."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val rhsOrtResultFile by option(
+        "--rhs-ort-result-file",
+        help = "The ORT result containing the left-hand-side scan result."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val outputOrtResultFile by option(
+        "--output-ort-result-file",
+        help = "The ORT result containing the left-hand-side scan result."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val lhsOrtResult = lhsOrtResultFile.readValue<OrtResult>()
+        val rhsOrtResult = rhsOrtResultFile.readValue<OrtResult>()
+
+        val rhsScanSummaries = rhsOrtResult.scanner!!.results.scanResults.flatMap { it.results }.associateBy(
+            keySelector = { it.provenance.key() },
+            valueTransform = { it.summary }
+        )
+
+        val scanResultContainers = lhsOrtResult.scanner!!.results.scanResults.mapTo(sortedSetOf()) { container ->
+            val results = container.results.map { lhsScanResult ->
+                val lhsSummary = lhsScanResult.summary
+                val rhsSummary = rhsScanSummaries[lhsScanResult.provenance.key()]
+
+                lhsScanResult.copy(summary = lhsSummary.minus(rhsSummary))
+            }
+            container.copy(results = results)
+       }
+
+       val result = lhsOrtResult.copy(
+           scanner = lhsOrtResult.scanner!!.copy(
+               results = lhsOrtResult.scanner!!.results.copy(
+                    scanResults = scanResultContainers
+               )
+           )
+       )
+
+       outputOrtResultFile.mapper().writerWithDefaultPrettyPrinter().writeValue(outputOrtResultFile, result)
+    }
+}
+
+private fun ScanSummary.minus(other: ScanSummary?): ScanSummary {
```

Comment:
How about [overloading an operator](https://kotlinlang.org/docs/reference/operator-overloading.html) for this, just for the fun of it and syntactic sugar?

Reply:
Done.